src/lib/tooltip/tooltip.ts

```diff
@@ -51,145 +75,209 @@ export class MdTooltip {
+
+  /** Subject for notifying that the tooltip has been hidden from the view */
+  private _onHide: Subject<any> = new Subject();
 
-      let portal = new ComponentPortal(TooltipComponent, this._viewContainerRef);
-      let tooltipRef = this._overlayRef.attach(portal);
-      tooltipRef.instance.message = this.message;
-      this._updatePosition();
+  /** Shows the tooltip with an animation originating from the provided origin */
+  show(position: TooltipPosition): void {
+    this._closeOnInteraction = false;
+    this._visibility = 'visible';
+    this._setTransformOrigin(position);
+
+    // Cancel the delayed hide if it is scheduled
+    if (this._hideTimeoutId) {
+      clearTimeout(this._hideTimeoutId);
     }
+
+    // If this was set to true immediately, then the body click would trigger interaction and
+    // close the tooltip right after it was displayed.
+    setTimeout(() => { this._closeOnInteraction = true; }, 0);
   }
 
-  /**
-   * Hides the tooltip and returns a promise that will resolve when the tooltip is hidden
-   */
-  hide(): void {
-    if (this.visible && this._overlayRef && this._overlayRef.hasAttached()) {
-      this.visible = false;
-      this._overlayRef.detach();
+  /** Begins the animation to hide the tooltip after the provided delay in ms */
+  hide(delay: number): void {
+    this._hideTimeoutId = setTimeout(() => {
+      this._visibility = 'hidden';
+      this._closeOnInteraction = false;
+    }, delay);
+  }
+
+  /** Returns an observable that notifies when the tooltip has been hidden from view */
+  afterHidden(): Observable<void> {
+    return this._onHide.asObservable();
+  }
+
+  /** Whether the tooltip is being displayed */
+  isVisible(): boolean {
+    return this._visibility === 'visible';
+  }
+
+  /** Sets the tooltip transform origin according to the tooltip position */
+  _setTransformOrigin(value: TooltipPosition) {
+    switch (value) {
+      case 'before': this._transformOrigin = 'right'; break;
+      case 'after':  this._transformOrigin = 'left'; break;
+      case 'above':  this._transformOrigin = 'bottom'; break;
+      case 'below':  this._transformOrigin = 'top'; break;
     }
   }
 
-  /**
-   * Shows/hides the tooltip and returns a promise that will resolve when it is done
-   */
-  toggle(): void {
-    if (this.visible) {
-      this.hide();
-    } else {
-      this.show();
+  _afterVisibilityAnimation(e: AnimationTransitionEvent): void {
+    if (e.toState === 'hidden' && !this.isVisible()) {
+      this._onHide.next();
+
     }
   }
 
   /**
-   * Updates the tooltip's position
+   * Interactions on the HTML body should close the tooltip immediately as defined in the
+   * material design spec.
+   * https://material.google.com/components/tooltips.html#tooltips-interaction
    */
-  private _updatePosition() {
-    if (this._overlayRef) {
-      this._changeDetectionRef.detectChanges();
-      this._overlayRef.updatePosition();
+  _handleBodyInteraction(): void {
```

Comment:
In practice, this is happens because the parent will create an overlay w/ tooltip on show, and detach the tooltip when the tooltip finishes the hide animation. Would you like this behavior to be implemented explicitly in the tooltip component?